src/common/libjob/job.c

```diff
@@ -25,26 +25,75 @@
 #if HAVE_CONFIG_H
 #include "config.h"
 #endif
-
+#include <unistd.h>
+#include <sys/types.h>
 #include <flux/core.h>
-
+#if HAVE_FLUX_SECURITY
+#include <flux/security/sign.h>
+#endif
 #include "job.h"
+#include "sign_none.h"
 
+#if HAVE_FLUX_SECURITY
+static flux_security_t *get_security_ctx (flux_t *h)
+{
+    const char *auxkey = "flux::job_security_ctx";
+    flux_security_t *sec = flux_aux_get (h, auxkey);
+    if (!sec) {
+        if (!(sec = flux_security_create (0)))
+            goto error;
+        if (flux_security_configure (sec, NULL) < 0)
+            goto error;
+        flux_aux_set (h, auxkey, sec, (flux_free_f)flux_security_destroy);
+    }
+    return sec;
+error:
+    flux_security_destroy (sec);
+    return NULL;
+}
+#endif
 
-flux_future_t *flux_job_submit (flux_t *h, const char *J, int flags)
+flux_future_t *flux_job_submit (flux_t *h, const char *jobspec, int priority,
+                                int flags)
 {
     flux_future_t *f;
+    const char *J;
+    char *s = NULL;
+    int saved_errno;
 
-    if (!h || !J) {
+    if (!h || !jobspec) {
         errno = EINVAL;
         return NULL;
     }
+    if (!(flags & FLUX_JOB_PRE_SIGNED)) {
+#if HAVE_FLUX_SECURITY
+        flux_security_t *sec = get_security_ctx (h);
+        if (!sec)
+            goto error;
+        if (!(J = flux_sign_wrap (sec, jobspec, strlen (jobspec), NULL, 0)))
+            goto error;
```

Comment:
If this fails, can you grab/set the errnum and grab/log the errmsg from the security context?
```C
       if (!(J = flux_sign_wrap (sec, jobspec, strlen (jobspec), NULL, 0))) {
           flux_log (h, LOG_ERR, "%s: %s", __FUNCTION__, flux_security_last_error (sec));
           errno = flux_security_last_errnum (sec);
           goto error;
       }
```

Reply:
Good point.  Certainly the errno should pass through if it doesn't already.   We shouldn't log from a library and there is currently no way to return an error string to the user since we are not returning a future.  I suppose we could go ahead and return a future and immediately fulfill it with an error and error string in that case (also if setting up the security context fails e.g. due to bad config file).

Hmm pondering...